lib/improver/nbhood/recursive_filter.py

```diff
@@ -318,8 +303,13 @@ def set_alphas(self, cube, alpha, alphas_cube):
         """
 
         if alphas_cube is None:
-            alphas_cube = cube.copy(
-                data=np.ones(cube.data.shape) * alpha)
+            if alpha is None:
+                emsg = ("A value for alpha must be set if alphas_cube is "
+                        "set to None: alpha is currently set as: {}")
+                raise ValueError(emsg.format(alpha))
+            else:
```

Comment:
As an interesting aside, we have generally been avoiding the use of these kinds of else statements. As you mentioned, indentation can become a pain when you end up several levels deep. In this case, if the exception is raised the code exits, if not it continues. So the else is redundant and the continuation of the code can be indented at the same level as the if alpha is None statement. (Basically delete the else and move the code back a level).